lib/apiservers/engine/backends/container.go

```diff
@@ -506,3 +653,235 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
 
 	return host, nil
 }
+
+func (c *Container) getImageMetadataFromStoragePL(image string) (*viccontainer.VicContainer, error) {
+	// FIXME: This is a temporary workaround until we have a name resolution story.
+	// Call imagec with -resolv parameter to learn the name of the vmdk and put it into in-memory map
+	cmdArgs := []string{"-reference", image, "-resolv", "-standalone", "-destination", os.TempDir()}
+
+	out, err := exec.Command(Imagec, cmdArgs...).Output()
+	if err != nil {
+		log.Printf("%s exit code: %s", Imagec, err)
+		return nil,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Container look up failed"),
+				http.StatusInternalServerError)
+	}
+	var v1 struct {
+		ID string `json:"id"`
+		// https://github.com/docker/engine-api/blob/master/types/container/config.go
+		Config container.Config `json:"config"`
+	}
+	if err := json.Unmarshal(out, &v1); err != nil {
+		return nil,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Failed to unmarshall image history: %s", err),
+				http.StatusInternalServerError)
+	}
+	log.Printf("v1 = %+v", v1)
+
+	imageMetadata := &viccontainer.VicContainer{
+		ID:     v1.ID,
+		Config: &v1.Config,
+	}
+
+	return imageMetadata, nil
+}
+
+func attachStreams(name string, tty, stdinOnce bool, clStdin io.ReadCloser, clStdout, clStderr io.Writer, keys []byte) error {
+	var wg sync.WaitGroup
+
+	errors := make(chan error, 3)
+	//FIXME: Swagger will timeout on us.  We need to either have an infinite timeout or the timeout should
+	// start after some inactivity?
+	inContext, inCancel := context.WithTimeout(context.Background(), 2*time.Hour)
+	outContext, outCancel := context.WithTimeout(context.Background(), 2*time.Hour)
+	errContext, errCancel := context.WithTimeout(context.Background(), 2*time.Hour)
+
+	// Create writers that will redirect copying to the writer we really want to
+	// write to.  These redirect writers will allow us to close
+	if clStdin != nil {
+		wg.Add(1)
```

Comment:
`Add()` in the go routine itself.  That's the idiom i tend to see and it makes for more readable code.  Also it removes this `if` condition.  Same comment for the 2 other instances of this below.